app/models/vm_scan.rb

```diff
@@ -492,26 +498,22 @@ def call_abort_retry(*args)
     end
   end
 
-  def process_delete_evm_snapshot(vm)
-    unless context[:snapshot_mor].nil?
-      mor = context[:snapshot_mor]
-      context[:snapshot_mor] = nil
-      set_status("Deleting snapshot before cancelling / aborting job")
-      if vm.kind_of?(ManageIQ::Providers::Openstack::CloudManager::Vm)
-        vm.ext_management_system.vm_delete_evm_snapshot(vm, mor)
-      elsif vm.kind_of?(ManageIQ::Providers::Microsoft::InfraManager::Vm) ||
-            (vm.kind_of?(ManageIQ::Providers::Azure::CloudManager::Vm) && vm.require_snapshot_for_scan?)
-        vm.ext_management_system.vm_delete_evm_snapshot(vm, :snMor => mor)
-      else
-        delete_snapshot(mor)
-      end
-    end
-  end
-
   def process_abort(*args)
     begin
       vm = VmOrTemplate.find_by(:id => target_id)
-      process_delete_evm_snapshot(vm)
+      unless context[:snapshot_mor].nil?
+        mor = context[:snapshot_mor]
+        context[:snapshot_mor] = nil
+        set_status("Deleting snapshot before aborting job")
+        if vm.kind_of?(ManageIQ::Providers::Openstack::CloudManager::Vm)
```

Comment:
Shouldn't you change this whole conditional block to:
```ruby
delete_snapshot(mor, vm)
```
now that `delete_snapshot` performs those checks?